### ISSUE TITLE
Event not showing up on the calendar

### ISSUE THREAD

**Issue by bodamithun** [builttoroam/device_calendar#221]
hey I am using the latest version in flutter app, trying to create an event. I see that the code return success, but when I go into the calendar, I see that event is not created. I would really appreciate any pointers as to why I am not seeing events in the calendar.

in a code  I am setting the calendar in a method, and calling another method to create the event.

           calendarsResult = await _deviceCalendarPlugin.retrieveCalendars();
           setState(() {
            _selectedCalendar = calendarsResult?.data[0];
            _addEventsToCalendar(_academicEvents);
           });

    Future _addEventsToCalendar(List<AcademicEvent> events) async {
        if (_selectedCalendar != null) {
         if (events != null && events.length > 1) {
           for (var event in events) {
          final eventTime = new DateTime(event.eventStartDate.year, event.eventStartDate.month,   event.eventStartDate.day);
          final eventToCreate = new Event(_selectedCalendar.id);
          eventToCreate.title = event.eventSummary;
          eventToCreate.start = eventTime;
          eventToCreate.description = event.eventSummary;
          ;
          String eventId = event.eventUID.toString();
          if (eventId != null) {
            eventToCreate.eventId = eventId;
          }
          eventToCreate.end = new DateTime(DateFormat.yMMMd().format(eventStartDate));
          final createEventResult =
              await _deviceCalendarPlugin.createOrUpdateEvent(eventToCreate);
          if (createEventResult.isSuccess &&
              (createEventResult.data?.isNotEmpty ?? false)) {
            //print("events added to local calander");
          }
        }
      }
    }

**Comment by nickrandolph** [comment]
Which platform are you seeing this on?

**Comment by bodamithun** [comment]
I am building a flutter app, in android studio.

**Comment by nickrandolph** [comment]
yes, but which platform (iOS or Android, or both) are you seeing this issue with?

**Comment by bodamithun** [comment]
oh..Both the platforms... calendar type is gmail

**Comment by nickrandolph** [comment]
If you query the calendar using the API do you see the event?

**Comment by bodamithun** [comment]
do you mean after adding the event to the calendar?

**Comment by nickrandolph** [comment]
yes - check that the API isn't seeing the event. Or is that what you meant by checking the calendar?

Also, please remove the line that sets the eventId - I suspect that might be preventing the event from being saved since this is auto generated.

**Comment by bodamithun** [comment]
I removed the id statement.

so this is interesting  in the createOrUpdateEvent(Event event), I see that the event is loaded.
but  when its invokes this method, at this line
  await channel.invokeMethod('createOrUpdateEvent', mapobject);
channel is null...

but it does not through any exceptions also..

**Comment by bodamithun** [comment]
ok... I changed the calendar to look at the default one and also removed the setting of id...and now it seems to be adding the event. but there is  another issue..

When ever I go to certain page, I am loading and adding events, but I see that the event is being added to the calendar every time I open that page, i was expecting it to add only once and update it if there is change. how does it determine if the event is unique or not? I removed setting the eventId based on your previous comment. so how would it determine if its new event or old one now?

**Comment by nickrandolph** [comment]
At this stage you'll need to retrieve the events and see if there is one that matches. You could use the Url property to store your event id and that way you can check the Url property

**Comment by bodamithun** [comment]
I see that retrieveevents functionality, but I see that the eventid is also required to be passed..so I am confused on how to retrieve events.

**Comment by KishanBusa8** [comment]
Hey i am also facing the same issue, But i figured it out where is the problem. 
Problem is when we use build app(after flutter build apk) it cant gave us access to use user's calendar, we cant retrieve any calendars from user's devices. But when we use it with same code at debugging time it grant us to access to user's devices calendar there's no issue.
I dont know how is this happen.

**Comment by nickrandolph** [comment]
@KishanBusa8 is this permissions related - are there permissions that need to be added to the manifest file perhaps?

**Comment by andzejsw** [comment]
Is this still an issue!?

**Comment by ni3vspace** [comment]
event has been created successfully, I don't see it in my calendar
